dask/dataframe/shuffle.py

```diff
@@ -684,9 +723,16 @@ def shuffle_group(df, col, stage, k, npartitions, ignore_index):
 
 
 def shuffle_group_3(df, col, npartitions, p):
-    g = df.groupby(col)
-    d = {i: g.get_group(i) for i in g.groups}
-    p.append(d, fsync=True)
+    try:
+        g = df.groupby(col)
+        d = {i: g.get_group(i) for i in g.groups}
+        p.append(d, fsync=True)
+    except Exception:
+        try:
+            p.drop()
```

Comment:
Why do we want to drop this?

Reply:
I think this matches the behavior on master of a case where something in `shuffle_group_3` raises an exception. Previously, `p` would go out of scope and be garbage collected, which includes a call to `partd.File.drop`.

Because we're explicitly providing a path to create the partd.File now, we're responsible for cleaning up.